packages/components/src/components/cardcheckbox/cardcheckbox.component.ts

```diff
@@ -0,0 +1,157 @@
+import { CSSResult, html, nothing, PropertyValues } from 'lit';
+import { property } from 'lit/decorators.js';
+import styles from './cardcheckbox.styles';
+import { DisabledMixin } from '../../utils/mixins/DisabledMixin';
+import { TabIndexMixin } from '../../utils/mixins/TabIndexMixin';
+import Card from '../card/card.component';
+import { CHECK_MARK, DEFAULTS, SELECTION_TYPE } from './cardcheckbox.constants';
+import { SelectionType } from './cardcheckbox.types';
+
+/**
+ * cardcheckbox component extends `mdc-card` and supports checkbox selection interaction addtionally.
+ *
+ * While using this component within a form or group of cards, make sure cards are in a role = "checkbox-group".
+ * This card would have events for selected and unselected (similar to checkbox)
+ *
+ * **Note**: This is a single selection card i.e. interacting anywhere on the card would toggle the checked state.
+ * Make sure to pass only non-interactable elements within the slots.
+ *
+ * Make sure to pass the `card-title` mandatorily for this card.
+ *
+ * @tagname mdc-cardcheckbox
+ *
+ * @dependency mdc-icon
+ * @dependency mdc-staticcheckbox
+ * @dependency mdc-text
+ *
+ * @slot before-body - This slot is for passing the content before the body
+ * @slot body - This slot is for passing the text content for the card
+ * @slot after-body - This slot is for passing the content after the body
+ *
+ * @event click - (React: onClick) Event that gets dispatched when the card is clicked. It toggles the checked state.
+ * @event keydown - (React: onKeyDown) This event is dispatched when a key is pressed down on the card.
+ * It toggles the checked state when enter key is used.
+ * @event keyup - (React: onKeyUp) This event is dispatched when a key is released on the card.
+ * It toggles the checked state when space key is used.
+ * @event change - (React: onChange) Event that gets dispatched when the card state changes.
+ * @event focus - (React: onFocus) Event that gets dispatched when the card receives focus.
+ *
+ */
+class CardCheckbox extends DisabledMixin(TabIndexMixin(Card)) {
+ /**
+   * The checked state of the card
+   * @default false
+   */
+ @property({ type: Boolean, reflect: true })
+ checked: boolean = false;
+
+ /**
+   * The selection type of the card. It can either be set to 'check' or 'checkbox'
+   * @default 'check'
+   */
+ @property({ type: String, attribute: 'selection-type' })
+ selectionType: SelectionType = DEFAULTS.SELECTION_TYPE;
+
+ constructor() {
+   super();
+   this.addEventListener('click', this.toggleChecked);
+   this.addEventListener('keydown', this.toggleOnEnter);
+   this.addEventListener('keyup', this.toggleOnSpace);
```

Comment:
Why to register both keydown and up events?
Can't we do something like this.

```suggestion
   this.addEventListener('keydown', this.handleKeyboardNavigation);
```

and in the component below,
```
private handleKeyboardNavigation(event: KeyboardEvent) {
     if (event.key === 'Enter' || event.key === ' ') {
          this.toggleChecked();
     }
}

Reply:
If you notice, they are 2 different functions that fire toggle on different keys.

On keydown, the toggle is fired through "enter" key. 
On keyup, toggle is fired using "space" key.

This is how the native button as well as mdc-buttonSimple behaves. Hence, adding the same behavior over here.